src/main/java/net/dv8tion/jda/bot/sharding/DefaultShardManagerBuilder.java

```diff
@@ -0,0 +1,661 @@
+     *
+     * @see    DefaultShardManager#addEventListener(Object...) JDA.addEventListener(Object...)
+     */
+    public DefaultShardManagerBuilder addEventListeners(final Object... listeners)
+    {
+        return this.addEventListeners(Arrays.asList(listeners));
+    }
+
+    /**
+     * Adds all provided listeners to the list of listeners that will be used to populate the {@link DefaultShardManager DefaultShardManager} object.
+     * <br>This uses the {@link net.dv8tion.jda.core.hooks.InterfacedEventManager InterfacedEventListener} by default.
+     * <br>To switch to the {@link net.dv8tion.jda.core.hooks.AnnotatedEventManager AnnotatedEventManager},
+     * use {@link #setEventManager(net.dv8tion.jda.core.hooks.IEventManager) setEventManager(new AnnotatedEventManager())}.
+     *
+     * <p><b>Note:</b> When using the {@link net.dv8tion.jda.core.hooks.InterfacedEventManager InterfacedEventListener} (default),
+     * given listener(s) <b>must</b> be instance of {@link net.dv8tion.jda.core.hooks.EventListener EventListener}!
+     *
+     * @param  listeners
+     *         The listener(s) to add to the list.
+     *
+     * @return The {@link net.dv8tion.jda.bot.sharding.DefaultShardManagerBuilder DefaultShardManagerBuilder} instance. Useful for chaining.
+     *
+     * @see    DefaultShardManager#addEventListener(Object...) JDA.addEventListener(Object...)
+     */
+    public DefaultShardManagerBuilder addEventListeners(final Collection<Object> listeners)
+    {
+        Checks.noneNull(listeners, "listeners");
+
+        this.listeners.addAll(listeners);
+        return this;
+    }
+
+    /**
+     * Builds a new {@link net.dv8tion.jda.bot.sharding.ShardManager ShardManager} instance and uses the provided token to start the login process.
+     * <br>The login process runs in a different thread, so while this will return immediately, {@link net.dv8tion.jda.bot.sharding.ShardManager ShardManager} has not
+     * finished loading, thus many {@link net.dv8tion.jda.bot.sharding.ShardManager ShardManager} methods have the chance to return incorrect information.
+     * <br>The main use of this method is to start the JDA connect process and do other things in parallel while startup is
+     * being performed like database connection or local resource loading.
+     *
+     * @throws  LoginException
+     *          If the provided token is invalid.
+     * @throws  IllegalArgumentException
+     *          If the provided token is empty or null.
+     * @throws  RateLimitedException
+     *          If we are being Rate limited.
+     *
+     * @return A {@link net.dv8tion.jda.bot.sharding.ShardManager ShardManager} instance that has started the login process. It is unknown as
+     *         to whether or not loading has finished when this returns.
+     */
+    public ShardManager buildAsync() throws LoginException, IllegalArgumentException, RateLimitedException
```

Comment:
This should be at the bottom

Reply:
Is there no buildBlocking? Why call it buildAsync in the first place then?

Reply:
There used to be `buildBlocking()` but it was a really bad implementation. Additionally having a method that can block up to for 30 minutes (and potentially even way more depending on amount of shard) sounds like a bad idea to me. 
I think the best idea is to just rename `buildAsync()` to just `build()`.